cpp/src/strings/search/find_instance.cu

```diff
@@ -0,0 +1,140 @@
+  size_type char_pos   = max_pos;
+  size_type char_count = 0;
+  size_type count      = 0;
+  for (auto itr = begin + lane_idx; itr + d_target.size_bytes() <= end;
+       itr += cudf::detail::warp_size) {
+    size_type const is_char = !is_utf8_continuation_char(*itr);
+    size_type const found   = is_char && (d_target.compare(itr, d_target.size_bytes()) == 0);
+    // count of threads that matched in this warp and produce an offset in each thread
+    auto const found_count = cg::reduce(warp, found, cg::plus<size_type>());
+    auto const found_scan  = cg::inclusive_scan(warp, found);
+    // handy character counter for threads in this warp
+    auto const chars_scan = cg::exclusive_scan(warp, is_char);
+    // activate the thread where we hit the desired find instance
+    auto const found_pos = (found_scan + count) == (instance + 1) ? chars_scan : char_pos;
+    // copy the position value for that thread into all warp threads
+    char_pos = cg::reduce(warp, found_pos, cg::less<size_type>());
+    if (char_pos < max_pos) { break; }  // all threads will stop
+    count += found_count;               // otherwise continue with the next set
+    char_count += cg::reduce(warp, is_char, cg::plus<size_type>());
+  }
+
+  // output the position if an instance match has been found
+  if (lane_idx == 0) { d_results[str_idx] = char_pos == max_pos ? -1 : char_pos + char_count; }
+}
+
+}  // namespace
+
+std::unique_ptr<column> find_instance(strings_column_view const& input,
+                                      string_scalar const& target,
+                                      size_type instance,
+                                      rmm::cuda_stream_view stream,
+                                      rmm::device_async_resource_ref mr)
+{
+  CUDF_EXPECTS(
+    instance >= 0, "Parameter instance must be positive integer or zero.", std::invalid_argument);
+  CUDF_EXPECTS(target.is_valid(stream), "Parameter target must be valid.", std::invalid_argument);
+
+  // create output column
+  auto results = make_numeric_column(data_type{type_to_id<size_type>()},
+                                     input.size(),
+                                     cudf::detail::copy_bitmask(input.parent(), stream, mr),
+                                     input.null_count(),
+                                     stream,
+                                     mr);
+  // if input is empty or all-null then we are done
+  if (input.size() == input.null_count()) { return results; }
+
+  auto d_target  = target.value(stream);
+  auto d_strings = column_device_view::create(input.parent(), stream);
+  auto d_results = results->mutable_view().data<size_type>();
+
+  constexpr thread_index_type block_size = 256;
+  constexpr thread_index_type warp_size  = cudf::detail::warp_size;
+  static_assert(block_size % warp_size == 0, "block size must be a multiple of warp size");
+  cudf::detail::grid_1d grid{input.size() * warp_size, block_size};
```

Comment:
```suggestion
  cudf::detail::grid_1d grid{input.size() * static_cast<int64_t>(warp_size), block_size};
```

Reply:
`warp_size` here is already int64_t (from line 116).

Reply:
Oh yes, I missed that alias line 👍